Examples/Algorithms/Geant4/src/Geant4Simulation.cpp

```diff
@@ -164,10 +105,15 @@ ActsExamples::ProcessCode ActsExamples::Geant4Simulation::execute(
 
   ACTS_DEBUG("Sending Geant RunManager the BeamOn() command.");
   // Start simulation. each track is simulated as a separate Geant4 event.
-  m_cfg.runManager->BeamOn(1);
+  runManager->BeamOn(1);
 
   // Since these are std::set, this ensures that each particle is in both sets
-  assert(eventData.particlesInitial.size() == eventData.particlesFinal.size());
+  if (eventData.particlesInitial.size() != eventData.particlesFinal.size()) {
```

Comment:
Hmm why turning this into a warning? This indicates severe logical issue in the code from my perspective and should lead to immediate failure.

Reply:
I will change it to `ACTS_ERROR`